core/src/main/java/io/parsingdata/metal/data/callback/BaseCallback.java

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright 2013-2016 Netherlands Forensic Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.parsingdata.metal.data.callback;
+
+import io.parsingdata.metal.data.Environment;
+import io.parsingdata.metal.data.ParseResult;
+import io.parsingdata.metal.token.Token;
+
+public class BaseCallback implements Callback {
```

Comment:
It is tested, but because it is not abstract, coverage requires a test that doesn't inherit from `BaseCallback` in order to execute the `handleSuccess()` and `handleFailure()` methods. I've reverted it to an abstract class.

Clients can trivially recreate the adapter pattern if they need it. Providing an abstract class seems cleaner for Metal I think.